client/state/initial-state.js

```diff
@@ -178,7 +177,10 @@ function getInitialPersistedState( initialReducer, currentUserId ) {
 			);
 
 			clearPersistedState();
-			return null;
+
+			// If state was randomly cleared save a flag indicating this
+			const isStateRandomlyCleared = ! config.isEnabled( 'force-sympathy' );
```

Comment:
It seems like `isStateRandomlyCleared` mirrors whether the sympathy mode is enabled, and not whether persisted state was cleared as a consequence of sympathy mode running.

Reply:
Actually it does mean that. If you look a few lines above this is where the state was cleared.
https://github.com/Automattic/wp-calypso/blob/bb7777c98f1ee12d4ddb40009d89081a1ab4d18d/client/state/initial-state.js#L178-L182

If the `force-sympathy` flag wasn't enabled (which means the state will always be cleared), And the state was actually cleared. It means that the random clearing is in effect. The actual line of code lives here.
https://github.com/Automattic/wp-calypso/blob/bb7777c98f1ee12d4ddb40009d89081a1ab4d18d/client/state/initial-state.js#L59-L64

Should I probably add a comment here explaining this?

Reply:
I understand what you mean 👍 It makes sense to not show the notice if `force-sympathy` is specifically enabled.

Reply:
@dmsnell 
> in that case, why are we checking the flag at all instead of directly returning { persistedStateWasCleared: true }

A notice is already bit of a hindrance to DX. If a developer is deliberately enabling sympathy, we do not need to state the obvious by showing them a warning because obviously they know what they are doing.

The main point of having a warning like this was to provide some feedback IFF a state was cleared **randomly**. And this is the exact case I am trying to capture by checking the flag. And then and only then will we be providing a warning.

https://github.com/Automattic/wp-calypso/blob/e7af7f07ed4a6466d2be0b5e5ccd31dc1b6d4aee/client/state/initial-state.js#L184

In other words if the state was cleared randomly
1 - [shouldAddSympathy returns true](https://github.com/Automattic/wp-calypso/blob/e7af7f07ed4a6466d2be0b5e5ccd31dc1b6d4aee/client/state/initial-state.js#L174)
2 - [The force-sympathy feature flag should **not** have been set](https://github.com/Automattic/wp-calypso/blob/e7af7f07ed4a6466d2be0b5e5ccd31dc1b6d4aee/client/state/initial-state.js#L184)